fixed mock hardware for botwheel explorer

ErikParkerrr opened the pull request:
Goes along with this issue: https://github.com/odriverobotics/ros_odrive/issues/24

Pretty much everything was already setup in the official repo. The only thing missing was the command and state interfaces for the `mock_components/GenericSystem` plugin to recognize. I've assumed adding these will not break anything for the non-mock-hardware usage. If it will, it can be changed to be in an if statement. 

Cheers

liborw committed:
Add service to clear errors

Co-authored-by: Libor Wagner (wagnelib) <libor.wagner@cvut.cz>

ErikParkerrr committed:
fixed mock hardware for botwheel explorer

samuelsadok commented on odrive_botwheel_explorer/description/urdf/diffbot.ros2_control.xacro:
```diff
@@ -18,12 +18,25 @@
         <param name="node_id">0</param>
+        <command_interface name="velocity"/>
+        <command_interface name="position"/>
+        <command_interface name="torque"/>
```
`torque` needs to be renamed to `effort` (4x). Otherwise, when I try it with the real hardware interface, it crashes with:
```
[ros2_control_node-1] terminate called after throwing an instance of 'std::runtime_error'
[ros2_control_node-1]   what():  Wrong state or command interface configuration.
[ros2_control_node-1] missing state interfaces:
[ros2_control_node-1] ' left_wheel_joint/torque '       ' right_wheel_joint/torque '
[ros2_control_node-1] missing command interfaces:
[ros2_control_node-1] ' left_wheel_joint/torque '       ' right_wheel_joint/torque '
```

samuelsadok reviewed (commented):
Ah cool, yeah this approach is preferred. The files are in part derived from the diffbot example (e.g. [this file](https://github.com/ros-controls/ros2_control_demos/blob/master/example_2/description/ros2_control/diffbot.ros2_control.xacro)) so this stub support was an artifact I wasn't aware of.

What command did you use to test this? When I tried it I saw that `botwheel_explorer.launch.py` needs some commented-out lines to be enabled to work with `use_mock_hardware:=true`. Please add these to the PR too.
![image](https://github.com/user-attachments/assets/d9ebd965-cecd-46f3-9e41-0b6d6f6bdce6)

ErikParkerrr committed:
changed torque to effort to match hardware interface

ErikParkerrr committed:
added use_mock_hardware and use_rviz launch arguemtns back in

ErikParkerrr committed:
added materials file from ros2_control_examples

ErikParkerrr committed:
Merge pull request #1 from ErikParkerrr/pr-28

Pr 28

ErikParkerrr commented:
- Fixed the issue you mentioned. 
- Allowed for the mock hardware argument to be passed through the launch file instead of editing the urdf directly like I was doing.
- Un-commented the rviz launch argument and grabbed rviz file from ros2_control example
- Added the materials file from the example so the urdf doesn't make references to non existent materials

ErikParkerrr committed:
added transmission ratios

ErikParkerrr committed:
fixed unused variable compiler complaint

ErikParkerrr committed:
allowed for defaulted transmission ratio of 1 for when undefined in urdf

samuelsadok reviewed (commented):
Ok the launch command `ros2 launch odrive_botwheel_explorer botwheel_explorer.launch.py use_mock_hardware:=true` works now, however the original command from the docs (`ros2 launch odrive_botwheel_explorer botwheel_explorer.launch.py`) is still broken due to a typo.

Furthermore, please rebase the PR to the latest `main` branch so that the history/diff only shows commits/changes related to this PR. Currently it includes a bunch of changes from an unrelated PR.

CLAassistant commented:
[![CLA assistant check](https://cla-assistant.io/pull/badge/not_signed)](https://cla-assistant.io/odriverobotics/ros_odrive?pullRequest=28) <br/>Thank you for your submission! We really appreciate it. Like many open source projects, we ask that you all sign our [Contributor License Agreement](https://cla-assistant.io/odriverobotics/ros_odrive?pullRequest=28) before we can accept your contribution.<br/>**0** out of **2** committers have signed the CLA.<br/><br/>:x: liborw<br/>:x: ErikParkerrr<br/><sub>You have signed the CLA already but the status is still pending? Let us [recheck](https://cla-assistant.io/check/odriverobotics/ros_odrive?pullRequest=28) it.</sub>

ErikParkerrr committed:
supressed more stupid stuff

ErikParkerrr committed:
fixed typo

ErikParkerrr committed:
made it unstamped

ErikParkerrr committed:
Merge branch 'main' into fix/unused_buf_fix

ErikParkerrr committed:
Merge pull request #2 from ErikParkerrr/fix/unused_buf_fix

Fix/unused buf fix and transmission ratios

ErikParkerrr committed:
fixed merge errors

ErikParkerrr committed:
Merge branch 'main' of https://github.com/odriverobotics/ros_odrive into odriverobotics-main

ErikParkerrr committed:
Merge branch 'odriverobotics-main'

ErikParkerrr committed:
fixed more merge errors

samuelsadok commented:
Is this PR still active? The last push seems to contain a lot of unrelated changes.

If it's still active I would ask to please:
- rebase to keep the PR limited in scope and the git history clean
- address my previous review comment
- sign the CLA

ErikParkerrr committed:
added the ability to reverse axis, kind of useless actually

ErikParkerrr committed:
Merge branch 'main' of https://github.com/odriverobotics/ros_odrive into odriverobotics-main

ErikParkerrr committed:
Merge branch 'odriverobotics-main'

samuelsadok commented:
@ErikParkerrr I'm closing this PR to avoid noise because the branch seems to be used for everyday development. We'd still be interested to see this contribution, and encourage you to open a new PR with the relevant changes only.
